duckduckgo/Android
# Android autofill MVP implementation

Task/Issue URL: https://app.asana.com/0/488551667048375/1202573773848143/f

### Description
Adds the MVP implementation of autofill for login credentials. 

It's MVP because 
- the UI is unpolished, which will be tackled imminently
- it's restricted to _internal_ testers until UI is more polished

ℹ️ Autofill is only enabled if both:

- the device has a password/PIN/pattern set for it.
- it's from an internal build, or has authenticated using DUO

### Steps to test this PR

For first set of tests, use _internal_ build type.

_Saving a login_
- [x] Visit a site where you can register (e.g., imdb.com) and create a new account
- [x] Verify you are prompted to save the login credentials; do it

_Autofilling a login_
- [x] Log out, and revisit the login page
- [x] Tap on the username field; verify prompt to autofill is shown; dismiss it
- [x] Tap on the username field again; verify prompt is not shown again
- [x] Refresh page and tap on the username field; verify prompt to autofill is shown; 
- [x] Choose to autofill
- [x] Verify credentials are autofilled into the login form

_Credential Management UI_

- [x] Visit overflow, then settings, and choose Autofill
- [x] Verify you can see the login saved above
- [x] Tap on it and verify it has the right username, password and URL
- [x] Test it can be edited, 
- [x] Test the username and password can be copied
- [x] Finally, test it can be deleted 
- [x] Try logging in to that page again once deleted, and verify no login prompt shown


For next set of tests, uninstall and use _play_ build type.

- Verify there are no signs of autofill by default
    - [x] No autofill option from settings
    - [x] Doesn't offer to save credentials when logging in

- Enable autofill, authenticating using Duo (details in linked Asana task)
    - [x] Perform some of the steps above to confirm autofill now enabled and working as expected
